lib/pbench/test/unit/server/test_datasets_metadata.py

```diff
@@ -424,21 +425,33 @@ def test_put_set_errors(self, capinternal, monkeypatch, query_get_as, query_put_
                 "metadata": {
                     "global.dashboard.seen": False,
                     "global.dashboard.nested.dummy": True,
-                    "dataset.name": "foo",
+                    "user.test": 1,
                 }
             },
             "drb",
-            HTTPStatus.INTERNAL_SERVER_ERROR,
+            HTTPStatus.OK,
         )
+        assert response.json == {
+            "errors": {
+                "global.dashboard.nested.dummy": "Key 'nested' value for "
+                "'global.dashboard.nested.dummy' "
+                "in (3)|drb is not a JSON object"
+            },
+            "metadata": {
+                "global.dashboard.nested.dummy": None,
+                "global.dashboard.seen": False,
+                "user.test": 1,
+            },
+        }
```

Comment:
So, even though `global.dashboard.nested.dummy` doesn't exist and there is an error reported on the attempt to set it, we still return an entry for it in the "output" part of the response?  This seems...ambiguous:  we _can_ set metadata keys to have a value of `None` (or equivalent), right?  So, returning a key in `"metadata"`, does not mean much without checking to see if it's also in `"errors"`.

(It seems like we should omit error'd keys from the "output"...no?)

Reply:
I don't see how it makes much difference, and the extra mechanism (creating a "shadow list" of successful changes) didn't seem worthwhile. Note that a `GET` including `global.dashboard.nested.dummy` would return `None` as the key doesn't exist, so this is consistent.